src/Database/PostgreSQL/Simple/ToField.hs

```diff
@@ -318,14 +318,13 @@ instance ToField UUID where
 instance ToField JSON.Value where
     toField = toField . JSON.encode
 
--- | Convert a Haskell value to a JSON 'JSON.Value' using
--- 'JSON.toJSON' and convert that to a field using 'toField'.
+-- | Convert a Haskell value to JSON using 'JSON.toEncoding'.
 --
 -- This can be used as the default implementation for the 'toField'
 -- method for Haskell types that have a JSON representation in
 -- PostgreSQL.
 toJSONField :: JSON.ToJSON a => a -> Action
-toJSONField = toField . JSON.toJSON
+toJSONField = toField . JSON.encode
```

Comment:
Does' this work?

`toJSON` produces `Value`, `encode` produces `ByteString`. `toField` does different things to them.

Reply:
Yep. `toField @Value` is defined as `toField = toField . JSON.encode` on line 319. The previous implementation of `toJSONField` needlessly creates an intermediate `Value`. The `Aeson` newtype wrapper in this repository also uses `encode` directly: https://github.com/haskellari/postgresql-simple/blob/master/src/Database/PostgreSQL/Simple/Newtypes.hs#L46